### PR TITLE
tests(binary): add more tests for sync

### PR DESCRIPTION
**Pull request by ee7**
Let's add our first tests for `--exercise` and the interactive mode, with a user typing `y`/`n`.

I'll refactor this later to be less terrible.

### CODE REVIEW

File: tests/test_binary.nim [hunk 1]
```diff
@@ -94,6 +94,141 @@ proc testsForSync(binaryPath: string) =
         [warn] some exercises are missing test cases
       """.dedent(8) # Not `unindent`. We want to preserve the indentation of the list items.
 
+    test "a single-exercise `sync` without `--update` exits with 1 and prints the expected output":
```
Comment (ErikSchierboom) [inline tests/test_binary.nim:97]: This might be slightly much, but maybe be slightly more specific about the test case?

```suggestion
    test "a single-exercise `sync` without `--update` and missing test case exits with 1 and prints the expected output":
```

File: tests/test_binary.nim [hunk 2]
```diff
@@ -94,6 +94,141 @@ proc testsForSync(binaryPath: string) =
...
+
+      check outp == """
+        Checking exercises...
+        [warn] anagram: missing 1 test case
+               - detects two anagrams (03eb9bbe-8906-4ea0-84fa-ffe711b52c8b)
+        [warn] some exercises are missing test cases
+      """.dedent(8)
+
+    test "when passing multiple exercises, only the final exercise is acted upon":
+      # TODO: We should instead support multiple exercises being passed.
```
Comment (ErikSchierboom) [inline tests/test_binary.nim:109]: Actually, I don't think we necessarily need to support this. It seems like rather an edge case.

**Review by ErikSchierboom — APPROVED** [review]

File: tests/test_binary.nim (hunk 1 above)
Comment (ee7) [reply tests/test_binary.nim]: Good catch, thanks. The previous description implied that `sync` without `--update` always exits with 1.

I've pushed https://github.com/exercism/configlet/pull/406/commits/735138dc8b4 to this PR, which tries to clarify the other test names too.

#### Before that commit:
```
[Suite] sync
  [OK] a `sync` without `--update` exits with 1 and prints the expected output
  [OK] a single-exercise `sync` without `--update` exits with 1 and prints the expected output
  [OK] when passing multiple exercises, only the final exercise is acted upon
  [OK] single-exercise `sync --update --mode=include` exits with 0 and includes the expected test cases
  [OK] `git diff` shows the expected diff
  [OK] single-exercise `sync --update --mode=exclude` exits with 0 and excludes the expected test cases
  [OK] `git diff` shows the expected diff
  [OK] single-exercise `sync --update` with input of 'y' includes as expected
  [OK] `git diff` shows the expected diff
  [OK] single-exercise `sync --update` with input of 'n' excludes as expected
  [OK] `git diff` shows the expected diff
  [OK] single-exercise `sync --update` with input of 's' exits with 1
  [OK] `git diff` shows the expected diff
  [OK] `sync --update --mode=include` exits with 0 and includes the expected test cases
  [OK] `git diff` shows the expected diff
  [OK] after syncing, another `sync --update --mode=include` performs no changes
  [OK] after syncing, a `sync` without `--update` shows that exercises are up to date
  [OK] the `git diff` output is still the same
```

#### With that commit:
```
[Suite] sync, without --update
  [OK] multiple exercises with missing test cases: prints the expected output, and exits with 1
  [OK] a given exercise with a missing test case: prints the expected output, and exits with 1
  [OK] when passing multiple exercises, only the final exercise is acted upon

[Suite] sync, with --update
  [OK] -mi: includes a missing test case for a given exercise, and exits with 0
  [OK] the diff is as expected
  [OK] -me: excludes a missing test case for a given exercise, and exits with 0
  [OK] the diff is as expected
  [OK] -mc: includes a missing test case for a given exercise when the input is 'y', and exits with 0
  [OK] the diff is as expected
  [OK] -mc: excludes a missing test case for a given exercise when the input is 'n', and exits with 0
  [OK] the diff is as expected
  [OK] -mc: neither includes nor excludes a missing test case for a given exercise when the input is 's', and exits with 1
  [OK] the diff is as expected
  [OK] -mi: includes every missing test case when not specifying an exercise, and exits with 0
  [OK] the diff is as expected
  [OK] after updating, another update using -mi performs no changes, and exits with 0
  [OK] after updating, a `sync` without `--update` shows that exercises are up to date, and exits with 0
  [OK] the diff is still the same
```

File: tests/test_binary.nim (hunk 2 above)
Comment (ee7) [reply tests/test_binary.nim:109]: Agreed. It's not a priority for me - I'm mainly adding a test case while I'm here to remind myself of the current behavior.

That said, if the user **does** pass multiple `--exercise` arguments, the possible behaviors seem to be:
1. Act on the first one (and probably print a warning)
1. Act on the final one (and probably print a warning)
1. Fail with an error, saying the only one `--exercise` is allowed
1. Act on all of the given exercises

In which case I probably prefer option 4. I think it can sometimes be useful to the maintainer, and is the least surprising and least friction for somebody who does try multiple `--exercise` arguments. And maybe it's useful for scripting.

I don't like the current behavior of acting on just one exercise without a warning. But we can discuss later.
